vector/test.c

```diff
@@ -78,4 +78,8 @@ void test_vector_copy(){
   vector_free(copy_arr);
 }
 
-
+void test_vector_create_size_NULL_return(){
+  size_t max_size = (size_t)-1;
```

Comment:
`(size_t)-1` should return maximum value of `size_t` as it is an unsigned type...
However isn't the `SIZE_MAX` macro working? It is **much more readable**.